pkg/preparation/shards/model/shard.go

```diff
@@ -0,0 +1,114 @@
+package model
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/ipfs/go-cid"
+	"github.com/storacha/guppy/pkg/preparation/types"
+	"github.com/storacha/guppy/pkg/preparation/types/id"
+)
+
+// ShardState represents the state of a shard.
+type ShardState string
+
+const (
+	// ShardStateOpen indicates that a shard is still accepting new nodes.
+	ShardStateOpen ShardState = "open"
+	// ShardStateClosed indicates that a shard is no longer accepting nodes, but
+	// is not yet added to the space.
+	ShardStateClosed ShardState = "closed"
+	// ShardStateAdded indicates that a shard has been added to the space.
+	ShardStateAdded ShardState = "added"
```

Comment:
are we going to have a ShardStateError? or just re-open a shard if some piece of data is not read correctly on upload

just curious

Reply:
Not totally sure yet, but it should fall out from the next stage.